### PR TITLE
Adding copyright to albums #1838

### LINKED ISSUE #1838: [Enhancement] Allow setting copyright per picture/album.
### Discussed in https://github.com/LycheeOrg/Lychee/discussions/1837

<div type='discussions-op-text'>

<sup>Originally posted by **elschman** May 14, 2023</sup>
Hi,
we (sports club) do have a couple (4,500) of pictures taken from several different photographs. All are in Lychee at our own webspace but we do not own the copyright but we have to state the copyright per picture together with the picture.
Is there any supported way to fulfil this request ?</div>

## TODO

- [ ] : Create field in the base_album table
- [ ] : Add returned filed to AlbumResource
- [ ] : Display field in Front-end
- [ ] : Add editing possibility in Front-end
- [ ] : Add end point for editing copyright in Backend.

- [ ] : Same as above for Photo if per photo integration is considered.

### PR DESCRIPTION
**Pull request by ThanasisMpalatsoukas**
Fixes #1838

Bringing the data of copyright to the frontend and adding a new route on api.php to edit the copyright value.
for issue #1838. After getting this PR accepted i will work on the frontend side and push on frontend the rest

### CODE REVIEW

**Comment by ildyria** [comment]
Thank you for your pull request. :)

Could you please add two tests:
- one where someone adds a copyright
- one where someone removes the copyright.

Also what should be the default value when it is not present?

You may also consider adding some localization strings: Like Copyright, Set copyright etc.. in the `lang/xx/` folders.

File: app/Http/Requests/Album/SetAlbumCopyrightRequest.php
Comment (ildyria) [inline app/Http/Requests/Album/SetAlbumCopyrightRequest.php:33]: you may need to do this in two steps. 
get the value in a temporary variable to type it with `/** @var Album ... */` and then assign to `$this->album`

Not the most convenient, I know.

File: app/Http/Controllers/AlbumController.php
Comment (ildyria) [inline app/Http/Controllers/AlbumController.php:202]: You need to add a `@property` annotation to the BaseAlbumImpl and BaseAlbum description.

Comment (ThanasisMpalatsoukas) [reply app/Http/Controllers/AlbumController.php:202]: Will look into the above issues. I was thinking the default value would be null or empty string when there is not value present.

File: app/Http/Requests/Album/SetAlbumCopyrightRequest.php
Comment (ildyria) [inline app/Http/Requests/Album/SetAlbumCopyrightRequest.php]:
```suggestion
		$this->album = $this->albumFactory->findBaseAlbumOrFail(
```

**Comment by ThanasisMpalatsoukas** [comment]
@ildyria 
Should i use google translate for the translations that i don't know or should i create the keys on all files and leave them empty?

**Comment by ildyria** [comment]
> @ildyria Should i use google translate for the translations that i don't know or should i create the keys on all files and leave them empty?

add the keys (the tests will require it anyway) and don't bother with google translate, just put the English. If someone wants to do the translation for their language, we will take the PR :)
I will probably make a change on yours to support French (as it is my native language).

File: lang/fr/lychee.php
Comment (ildyria) [inline lang/fr/lychee.php]:
```suggestion
	'ALBUM_COPYRIGHT' => 'Droits d’auteur',
```

File: lang/fr/lychee.php
Comment (ildyria) [inline lang/fr/lychee.php]:
```suggestion
	'ALBUM_SET_COPYRIGHT' => 'Définir les droits d’auteur',
```

**Comment by ildyria** [comment]
@ThanasisMpalatsoukas  I fixed the conflicts with master.

We will merge this PR when:
- tests are added (see above)
- front-end supports it:
    - You will need to make a PR on Lychee-front for that
    - Once merged, we can rebuild the js files and we will be able to merge this one in master.
    
    
:smiley:

**Comment by ildyria** [comment]
@ThanasisMpalatsoukas any updates ? :(

**Comment by ThanasisMpalatsoukas** [comment]
@ildyria Hey, was busy with work and stuff. Will probably take a look at it this week :)

**Comment by ildyria** [comment]
> was busy with work and stuff.

I know that feeling quite well. :')

**Comment by ildyria** [comment]
@ThanasisMpalatsoukas any updates ? :(

**Comment by ARMeeru** [comment]
@ildyria Is it still pending?

**Comment by ildyria** [comment]
Well yes. Those points have still not been resolved.

> We will merge this PR when:
> 
> * tests are added (see above)
> * front-end supports it:
>   
>   * You will need to make a PR on Lychee-front for that
>   * Once merged, we can rebuild the js files and we will be able to merge this one in master.

Though what I would suggest is to fork this Pull request and start integrating against the Livewire branch of #1303 as it will make it significantly easier for the user to add tests and not have to care for the JS front side.

**Comment by ARMeeru** [comment]
I'm interested on this. While finalising that interest, I would like to ping @ThanasisMpalatsoukas in case he's on this but forgot.

**Comment by ThanasisMpalatsoukas** [comment]
@ARMeeru thank you for being considerate and pinging me on this, sadly i have too personal work and am unable to continue with this at this point in time. You can go on and finalize this feature :)

**Comment by ildyria** [comment]
@ARMeeru If you decide to go on this feature, I strongly advise you to have a look at https://github.com/LycheeOrg/Lychee/pull/2035 before.

You will probably want to check the following files:
https://github.com/LycheeOrg/Lychee/blob/livewire/app/View/Components/Gallery/Album/Details.php
https://github.com/LycheeOrg/Lychee/blob/livewire/resources/views/components/gallery/album/details.blade.php

https://github.com/LycheeOrg/Lychee/blob/livewire/app/Livewire/Components/Forms/Album/Properties.php
https://github.com/LycheeOrg/Lychee/blob/livewire/resources/views/livewire/forms/album/properties.blade.php

And associated tests:
https://github.com/LycheeOrg/Lychee/blob/livewire/tests/Livewire/Forms/Album/PropertiesTest.php

This should be a good read too: https://lycheeorg.github.io/docs/livewire.html

**Review by ildyria — APPROVED** [review]
Tested LGTM.

**Pull request merged**
